src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/HostFileChangeMonitor.cs

```diff
@@ -97,6 +97,13 @@ private static void InitFCN()
                 }
                 if (fcn == null)
                 {
+#if NET5_0_OR_GREATER
+                    if (OperatingSystem.IsBrowser())
+                    {
+                        throw new PlatformNotSupportedException();
```

Comment:
Isn't this method should have annotated with  `[UnsupportedOSPlatform("browser")]` ?

Reply:
No, because you could implement your own custom version of `IFileChangeNotificationSystem` that works on browser and pass it in via `ObjectCache.Host`